src/rashdf/geom.py

```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
+            multi_lines.append(
+                MultiLineString(
+                    list(points[part_pnt_start:part_pnt_start+part_pnt_cnt] for part_pnt_start, part_pnt_cnt in parts)
+                )
+            )
+        return GeoDataFrame(
+            {
+                "bc_line_id":bc_line_ids,
+                "name":names,
+                "mesh_name":mesh_names,
+                "type":types,
+                "geometry":multi_lines
+            },
+            geometry="geometry",
+            crs=self.projection()
+        )
 
     def breaklines(self) -> GeoDataFrame:
-        raise NotImplementedError
+        """Return the 2D mesh area breaklines.
+        
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the 2D mesh area breaklines if they exist.
+        """
+        if "/Geometry/2D Flow Area Break Lines" not in self:
+            return GeoDataFrame()
+        bl_line_data = self["/Geometry/2D Flow Area Break Lines"]
+        bl_line_ids = range(bl_line_data["Attributes"][()].shape[0])
+        names = np.vectorize(convert_ras_hdf_string)(bl_line_data["Attributes"][()]["Name"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bl_line_data["Polyline Info"][()]:
+            points = bl_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bl_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
+                    list(points[part_pnt_start:part_pnt_start+part_pnt_cnt] for part_pnt_start, part_pnt_cnt in parts)
+                )
+            )
+        return GeoDataFrame(
+            {
+                "bl_id":bl_line_ids,
+                "name":names,
+                "geometry":multi_lines
+            },
+            geometry="geometry",
+            crs=self.projection()
+        )
 
     def refinement_regions(self) -> GeoDataFrame:
-        raise NotImplementedError
+        """Return the 2D mesh area refinement regions.
+        
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the 2D mesh area refinement regions if they exist.
+        """
+        if "/Geometry/2D Flow Area Refinement Regions" not in self:
+            return GeoDataFrame()
+        rr_data = self["/Geometry/2D Flow Area Refinement Regions"]
+        rr_ids = range(rr_data["Attributes"][()].shape[0])
+        names = np.vectorize(convert_ras_hdf_string)(rr_data["Attributes"][()]["Name"])
+        multi_polygons = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in rr_data["Polygon Info"][()]:
+            points = rr_data["Polygon Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = rr_data["Polygon Parts"][()][part_start:part_start+part_cnt]
+            multi_polygons.append(
+                MultiLineString(
```

Comment:
Should this be `Polygon` instead?

Reply:
Thanks for catching this! Updated to `Polygon`/`MultiPolygon`.